src/Nethermind/Nethermind.Synchronization/Peers/ISyncPeerPool.cs

```diff
@@ -100,12 +100,26 @@ Task<SyncPeerAllocation> Allocate(
 
     public static class SyncPeerPoolExtensions
     {
-        public static async Task<T> AllocateAndRun<T>(
+        public static Task<T> AllocateAndRun<T>(
             this ISyncPeerPool syncPeerPool,
             Func<ISyncPeer, Task<T>> func,
             IPeerAllocationStrategy peerAllocationStrategy,
             AllocationContexts allocationContexts,
             CancellationToken cancellationToken)
+        {
+            return syncPeerPool.AllocateAndRun2(
+                (peerInfo) => func(peerInfo?.SyncPeer),
+                peerAllocationStrategy,
+                allocationContexts,
+                cancellationToken);
+        }
+
+        public static async Task<T> AllocateAndRun2<T>(
```

Comment:
confused about the naming, can we just name it `AllocateAndRun`?

Reply:
`AllocateAndRun` already used.

Reply:
but has different parameters?

Reply:
Yes, one is `Func<ISyncPeer>` another is `Func<PeerInfo>`.

Reply:
So we should be able to name them the same

Reply:
Ow I forgot this is c#